Source/Core/UICommon/AutoUpdate.h

```diff
@@ -18,6 +18,28 @@ class AutoUpdateChecker
 
   static bool SystemSupportsAutoUpdates();
 
+  enum class ShouldSilentlyFail
+  {
+    Yes,
+    No,
+  };
+
+  // Passed to the OnErrorOccurred callback.
+  enum class CheckError
+  {
+    // The HTTP request failed.
+    RequestFailed,
+
+    // The JSON received from the server is not valid.
+    InvalidJson,
+
+    // We're already up to date.
+    AlreadyUpToDate,
+
+    // Couldn't launch updater.
+    FailedToLaunchUpdater,
```

Comment:
IMO InvalidJsonInServerResponse is more readable (considering PascalCase is used and caps are used to separate words). We don't have particularly consistent rules on this though...

Reply:
I changed the enums and removed the comments to match your suggestions. For JSON vs Json, I went with Json since I agreed that it looked more readable. (I couldn't find an example elsewhere where an acronym is used in an enum, but I'm probably blind.)

I also changed RestartMode lower down to PascalCase to make the styling consistent within the file. There seems to be a bunch of other enums where ``WHATEVER_THIS_IS_CALLED`` styling is used, but PascalCase seems to be prevalent?

Reply:
SCREAMING_SNAKE_CASE, I think